lib/kino_maplibre/map_cell.ex

```diff
@@ -293,34 +306,68 @@ defmodule KinoMapLibre.MapCell do
         do: %{
           "source_id" => layer["layer_source"],
           "source_data" => layer["layer_source"],
-          "source_type" => layer["layer_source_type"]
+          "source_type" => layer["source_type"],
+          "source_coordinates" => source_coordinates(layer)
         },
         uniq: true
   end
 
+  defp source_coordinates(%{"source_type" => "table", "coordinates_format" => "columns"} = layer) do
+    {:lng_lat, [layer["source_longitude"], layer["source_latitude"]]}
+  end
+
+  defp source_coordinates(%{"source_type" => "table"} = layer) do
+    {String.to_atom(layer["coordinates_format"]), layer["source_coordinates"]}
+  end
+
+  defp source_coordinates(_), do: nil
+
+  defp add_source_function(:geo), do: :add_geo_source
+  defp add_source_function(:table), do: :add_table_source
+  defp add_source_function(_), do: :add_source
+
   defp missing_dep() do
     unless Code.ensure_loaded?(MapLibre) do
       ~s/{:maplibre, "~> 0.1.0"}/
     end
   end
 
-  defp new_layer(layer_source \\ nil, layer_source_type \\ nil) do
+  defp new_layer(layer_source \\ nil, source_type \\ nil) do
     [
       %{
         "layer_id" => nil,
         "layer_source" => layer_source,
-        "layer_source_type" => layer_source_type,
+        "source_type" => source_type,
         "layer_type" => "circle",
         "layer_color" => "black",
         "layer_opacity" => 1,
-        "layer_radius" => 10
+        "layer_radius" => 10,
+        "coordinates_format" => "lng_lat",
+        "source_coordinates" => nil,
+        "source_longitude" => nil,
+        "source_latitude" => nil
       }
     ]
   end
 
-  defp is_geometry(%module{}) when module in @geometries, do: true
-  defp is_geometry("http" <> url), do: url |> String.split(".") |> List.last() == "geojson"
-  defp is_geometry(_), do: false
+  defp is_geometry?(%module{}) when module in @geometries, do: true
+  defp is_geometry?("http" <> url), do: url |> String.split(".") |> List.last() == "geojson"
+  defp is_geometry?(_), do: false
+
+  defp is_table?(val), do: implements?(Table.Reader, val)
+
+  defp source_type(%module{}) when module in @geometries, do: "geo"
+  defp source_type(val), do: if(is_table?(val), do: "table")
+
+  defp columns_for(data) do
+    with true <- implements?(Table.Reader, data),
+         {_, %{columns: columns}, _} <- Table.Reader.init(data),
+         true <- Enum.all?(columns, &implements?(String.Chars, &1)) do
+      Enum.map(columns, &to_string/1)
```

Comment:
And in this case we probably want to reject everything that is not atom/string:

```suggestion
         {_, %{columns: columns}, _} <- Table.Reader.init(data) do
      for column <- columns, is_atom(column) or is_binary(column), do: column
```

If this pattern came from kino_vega_lite, we need to update kino_vega_lite too. :)

Reply:
@josevalim I see no problem allowing keys that are charlists or numbers, the real constraint is that they need to have string representation, so we can show them in the select. Any reason to skip those cases?

As for the the reasoning behind the `all?` check, if there's a non-stringable key, then the map is likely not a real table (other than structurally). Thoughts?